### ISSUE TITLE
2.3.2  -- Illegal reflective access

### ISSUE THREAD

**Issue by solveretur** [bolcom/spring-data-mongodb-encrypt#30]
Hi, I use your library and every time I start my application I encounter such log:
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.bol.util.JCEPolicy (file:/home/xxxx/.gradle/caches/modules-2/files-2.1/com.bol/spring-data-mongodb-encrypt/2.3.2/e3d41f51d1995306fb7a30f1230afcc18ea19e9d/spring-data-mongodb-encrypt-2.3.2.jar) to field javax.crypto.JceSecurity.isRestricted
WARNING: Please consider reporting this to the maintainers of com.bol.util.JCEPolicy
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

my java version:

```
openjdk 11.0.6 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10)
OpenJDK 64-Bit Server VM (build 11.0.6+10, mixed mode)
```
Please solve it - I really like your library and I'm afraid it will break in the future because of it :)

**Comment by agoston** [comment]
Don't worry. This is a known hack to enable the strong encryption in java8 versions before update 162. It won't break a thing.
I will add a java version check to only run this on java 8.

**Comment by agoston** [comment]
This has been fixed in 2.4.1.
Thanks for reporting!
